tests/test_scalecomposite.py

```diff
@@ -0,0 +1,314 @@
+                   response.aggregate().data(['sample', 'energy'],
+                                             sorted_by='num_occurrences',
+                                             reverse=True)]
+        self.assertIn(samples[0][0], [{'a': 1, 'b': -1}, {'a': -1, 'b': 1}])
+        self.assertAlmostEqual(samples[0][1], 1.8)
+
+    def test_sample_ising_ignore_offset(self):
+        linear = {'a': -4.0, 'b': -4.0}
+        quadratic = {('a', 'b'): 3.2}
+        offset = 5
+
+        sampler = ScaleComposite(SimulatedAnnealingSampler())
+        response = sampler.sample_ising(linear, quadratic, offset=offset,
+                                        scalar=0.5,
+                                        ignore_offset=True, num_reads=100)
+        samples = [(s, e) for s, e, in
+                   response.aggregate().data(['sample', 'energy'],
+                                             sorted_by='num_occurrences',
+                                             reverse=True)]
+        self.assertEqual({'a': 1, 'b': 1}, samples[0][0])
+        self.assertAlmostEqual(samples[0][1], 0.2)
+
+    def test_sample_hising(self):
+        linear = {'a': -4.0, 'b': -4.0, 'c': -4.0}
+        quadratic = {('a', 'b', 'c'): 3.2}
+        offset = 5
+        sampler = ScaleComposite(HigherOrderComposite(
+            SimulatedAnnealingSampler()))
+
+        response = sampler.sample_ising(linear, quadratic,
+                                        offset=offset, scalar=0.5,
+                                        num_reads=100,
+                                        penalty_strength=5.0,
+                                        )
+        samples = [(s, e) for s, e, in
+                   response.aggregate().data(['sample', 'energy'],
+                                             sorted_by='num_occurrences',
+                                             reverse=True)]
+        self.assertEqual({'a': 1, 'b': 1, 'c': 1}, samples[0][0])
+        self.assertAlmostEqual(samples[0][1], -3.8)
+
+    def test_sample_hising_ignore_variables(self):
+        linear = {'a': -4.0, 'b': -4.0, 'c': -4.0}
+        quadratic = {('a', 'b', 'c'): 3.2}
+        offset = 5
+        sampler = ScaleComposite(HigherOrderComposite(
+            SimulatedAnnealingSampler()))
+        response = sampler.sample_ising(linear, quadratic, scalar=0.5,
+                                        penalty_strength=5.0,
+                                        ignored_interactions=[
+                                            ('a', 'b', 'c')],
+                                        ignored_variables=['a'], num_reads \
+                                            =100)
+
+        samples = [(s, e) for s, e, in
+                   response.aggregate().data(['sample', 'energy'],
+                                             sorted_by='num_occurrences',
+                                             reverse=True)]
+        self.assertIn(samples[0][0], [{'a': 1, 'b': -1, 'c': 1},
```

Comment:
This is not guaranteed. `SimulatedAnnealingSampler` should not be used for testing